### PR TITLE
ENH k-space wrapping for Hermitian images

### PR DESCRIPTION
**Pull request by beckermr**
This PR has a very naive, but working implementation of k-space wrapping for Hermitian images. The code expands the image from its compressed format to the fully explicit Hermitian format, wraps then using simple code, and then cuts away the parts not needed. 

I tried a more memory efficient algorithm that didn't involve the expansion but could not get it working. I've left that code commented out in case someone else can find the bug. 

todo:
 - [x] add tests of reverse-mode autodiff

### CODE REVIEW

File: tests/jax/galsim/test_shear_position_jax.py [hunk 1]
```diff
@@ -125,6 +131,9 @@ def test_wrap_jax_complex():
     b = galsim.BoundsI(-K + 1, K, -L + 1, L)
     b2 = galsim.BoundsI(-K + 1, K, 0, L)
     b3 = galsim.BoundsI(0, K, -L + 1, L)
+    # print('b = ',b)
+    # print('b2 = ',b2)
+    # print('b3 = ',b3)
```
Comment (beckermr) [inline tests/jax/galsim/test_shear_position_jax.py:136]:
```suggestion
```

File: tests/jax/galsim/test_shear_position_jax.py
Comment (beckermr) [inline tests/jax/galsim/test_shear_position_jax.py:1]: This test file is redundant so I removed it.

File: tests/jax/test_image_wrapping.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+from galsim_test_helpers import timer
+
+from jax_galsim.core.wrap_image import (
+    expand_hermitian_x, expand_hermitian_y, contract_hermitian_x,
+    contract_hermitian_y,
+)
+
+
+@timer
+def test_image_wrapping():
```
Comment (beckermr) [inline tests/jax/test_image_wrapping.py:12]:
```suggestion
def test_image_wrapping_expand_contract():
```

File: jax_galsim/core/wrap_image.py [hunk 3]
```diff
@@ -33,3 +33,107 @@ def _body_i(i, vals):
...
+@jax.jit
+def wrap_hermitian_y(im, im_xmin, im_ymin, wrap_xmin, wrap_ymin, wrap_nx, wrap_ny):
+    im_exp = expand_hermitian_y(im)
+    im_exp = wrap_nonhermitian(
+        im_exp, wrap_xmin - im_xmin, wrap_ymin - im_ymin, wrap_nx, wrap_ny
+    )
+    return contract_hermitian_y(im_exp)
+
+
+# I am leaving this code here for posterity. It has a bug that I cannot find.
```
Comment (beckermr) [inline jax_galsim/core/wrap_image.py:76]: @rmjarvis @jecampagne If any of you have a chance to look for bugs in the code below, that'd be cool. No worries if not!

File: tests/jax/test_image_wrapping.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        2 * M + 1, N + 1, xmin=-M, ymin=0
+    )  # Implicitly Hermitian across y axis
+    im3 = galsim.ImageCD(
+        M + 1, 2 * N + 1, xmin=0, ymin=-N
+    )  # Implicitly Hermitian across x axis
+    # print('im = ',im)
+    # print('im2 = ',im2)
+    # print('im3 = ',im3)
+    b = galsim.BoundsI(-K + 1, K, -L + 1, L)
+    im_test = galsim.ImageCD(b, init_value=0)
```
Comment (ismael-mendoza) [inline tests/jax/test_image_wrapping.py:33]: what is the purpose of `im_test`? it doesn't seem like you test anything with it

**Review by ismael-mendoza — APPROVED** [review]
LGTM! just two questions below

File: tests/jax/test_image_wrapping.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    im_cnt = contract_hermitian_y(im.array)
+    np.testing.assert_allclose(
+        im_cnt,
+        im2.array,
+        err_msg="contract_hermitian_x() did not match expectation",
+    )
+
+
+@timer
+def test_image_wrapping_autodiff():
```
[the remark this answers is not in the document]
Comment (beckermr) [reply tests/jax/test_image_wrapping.py:89]: We should test both. Typical reverse has stricter requirements in the jax api.

File: tests/jax/test_image_wrapping.py (hunk 4 above)
Comment (beckermr) [reply tests/jax/test_image_wrapping.py:33]: Good catch. Hold over from the other test!

**Review by beckermr — COMMENTED** [review]
After working through the image interp pr I am worried there is a bug here. Going to hold off on merging.

**Comment by beckermr** [comment]
I am merging this code with the fixes to address the PR comments above into PR #60 since it is needed for tests there.

**Pull request closed without merging**